store/postgres/migrations/2022-03-09-07000_call_cache_block_number_idx/up.sql

```diff
@@ -0,0 +1,18 @@
+create index if not exists
+	eth_call_cache_block_number_idx
+on
+	eth_call_cache(block_number);
+
+do $$
+declare
+	tables cursor for select namespace from chains;
```

Comment:
One thing I just noticed: `chains` is the global directory of all chains. It is maintained only in the primary, though gets copied to the other shards periodically, but only as a fallback in case the primary database goes down. The list of chains in a specific shard is in `ethereum_networks`, and that table should be consulted for the index creation here.

Reply:
Thanks for catching this! Fixed in https://github.com/graphprotocol/graph-node/pull/3343/commits/b440fad40c8e605dceeaf76f17f6671601724dc4.